integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchemaCommon.scala

```diff
@@ -263,18 +258,48 @@ class AlterTableColumnSchemaGenerator(
       isVarcharColumn(childField.name.getOrElse(childField.column)))
   }
 
+  def addComplexChildCols(currField: Field,
+      currColumnSchema: ColumnSchema,
+      newCols: mutable.Buffer[ColumnSchema],
+      allColumns: mutable.Buffer[ColumnSchema],
+      longStringCols: mutable.Buffer[ColumnSchema],
+      currentSchemaOrdinal: Int): Unit = {
+    if (currField.children.get == null || currField.children.isEmpty) {
+      return
+    }
+    if (DataTypes.isArrayType(currColumnSchema.getDataType) ||
```

Comment:
can directly check if it is complextype. currColumnSchema.getDataType.iscomplextype

Reply:
Done